libraries/common/stringbuffer.effekt

```diff
@@ -35,7 +36,10 @@ def stringBuffer[A] { prog: => A / StringBuffer }: A = {
       resume(())
     }
     def flush() = {
+      // resize buffer to strip trailing zeros that otherwise would be converted into 0x00 characters
+      buffer = bytearray::resize(buffer, pos)
       val str = buffer.toString()
+      // after flushing, the stringbuffer should be empty again
       buffer = bytearray::allocate(initialCapacity)
```

Comment:
I think this could be made somewhat more efficient -- `resize` copies into a newly allocated bytearray, but `toString` doesn't (since on LLVM strings are bytearrays)
So it should be possible to just _clear_ the buffer without allocating it again?
```suggestion
      // resize buffer to strip trailing zeros that otherwise would be converted into 0x00 characters
      val str = bytearray::resize(buffer, pos).toString()
      // after flushing, the stringbuffer should be empty again
      buffer = buffer.fill(0)
```
where `buffer.fill(0)` means `memset(buffer, 0)` (which we don't have yet) :)

Reply:
Yes, you could just clear the buffer, however, then the buffer's size will not be reset to a (potentially) smaller size and instead just buffer's previous size is kept. This probably would not be a big deal, though.